### ISSUE TITLE
QSelect: 'required' attribute - unexpected behaviour

### ISSUE THREAD

**Issue by maddogcz** [quasarframework/quasar#12779]
### What happened?

When I use `required` attribute on QSelect, the behaviour is very unexpected:

- When used with `useInput`, the `required` attribute is passed to the input and the form cannot be submitted due to automatic HTML form validation
- When used without `useInput`, `required` is passed to some wrapper div and it doesn't affect the form at all

### What did you expect to happen?

Make `required` attribute comply with automatic HTML validation or omit it completely with proper disclaimer in the docs (because the automatic validation doesn't work when the element is not visible, which is the case of the actual `select`)

### Reproduction URL

https://codepen.io/maddogcz/pen/qBpEOwZ

### How to reproduce?

1. Value is set, click on submit, the second select throws validation error (should pass)
2. Select `None` value, click on submit - again the second select throws validation error (but this time it should have been the first one, because value is empty and both are required)

### Flavour

Vue CLI Plugin (vue-cli-plugin-quasar)

### Areas

Components (quasar)

**Comment by rstoenescu** [comment]
Hi,

There are a thousand attributes that you can specify for any component. When you have an input (like when using `use-input` prop) developers have requested that all the non-props to be passed on to the inner input -- which is what we did. When you're not using an input, it doesn't makes any sense to use "required" (as in "required" applied to what?).
But the main point: if you want validation, we have it (both baked in or the possibility to use an external one), but none imply any native browser behavior which is unpredictable cross-browsers anyway. So use one of the two validation methods. https://quasar.dev/vue-components/input#validation (yes, I know it's the QInput page, but it applies to QSelect too)

**Comment by septatrix** [comment]
I spend more time than I would like to admit before noticing that QSelect does not support the required attribute. While I personally have not experienced any cross-browser inconsistencies between `<select/input required>`, I can certainly understand that supporting `required` might be troublesome, especially together with `use-input` (what should it apply to then?).

Nonetheless I feel like this situation could be improved by doing the following (not exclusive):

1. Add a small `required` function to the [validation helpers](https://github.com/quasarframework/quasar/blob/dev/ui/src/utils/patterns.js) which simply checks if a value is present.
2. Add a rules/validation section to the `QSelect` docs. This would fit ideally in the "The model" section and could be e.g. require a selection in single mode, require a minimum count in multiple selection or even something more exotic like enforce selection of an even number of elements. This section could either refer to the helper from (1) or inline them if (1) is not implemented

**Comment by snupsband** [comment]
I ran into this same issue this morning and it makes me really hate Quasar.

- I made a QSelect required because I want the user to have to enter a value.
- I put a value in there and it won't validate.

This is confusing. The attribute seems to partially work.

I ended up doing this instead: `:rules="[validateUser]"`

I regret choosing Quasar for this project. Why does Quasar work so differently from every other UI toolkit I've ever used?
